src/python/zquantum/core/utils.py

```diff
@@ -439,3 +441,41 @@ def save_noise_model(noise_model_data, module_name, function_name, filename):
 
     with open(filename, "w") as f:
         f.write(json.dumps(data, indent=2))
+
+
+def create_symbols_map(
+    symbols: List[sympy.Symbol], params: np.ndarray
+) -> Tuple[sympy.Symbol, float]:
+    """
+    Creates a map to be used for evaluating sympy expressions.
+
+    Args:
+        symbols: list of sympy Symbols to be evaluated
+        params: numpy array containing numerical value for the symbols
+    """
+    if len(symbols) != len(params):
+        raise (
+            ValueError(
+                "Length of symbols: {0} doesn't match length of params: {1}".format(
+                    len(symbols), len(params)
+                )
+            )
+        )
+    return [(symbol, param) for symbol, param in zip(symbols, params.tolist())]
+
+
+def generate_random_params(
+    number_of_params: int, min_val: float = 0, max_val: float = 1.0
+) -> np.ndarray:
+    """For the given ansatz, generate random parameters.
+
+    Args:
+        number_of_params (int): the number of parameters to generate
+        min_val (float): minimum parameter value
+        max_val (float): maximum parameter value
+
+    Returns:
+        numpy.ndarray: the generated parameters
+    """
+    random_params = np.random.sample(number_of_params)
+    return (max_val - min_val) * random_params + min_val
```

Comment:
I might be missing something, but did we just reinvent `numpy.random.uniform`? :joy: 

https://docs.scipy.org/doc//numpy-1.15.0/reference/generated/numpy.random.uniform.html